File: Data Structures & Algorithms/first-missing-positive/submission-0.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        int n = nums.length;
        boolean[] seen = new boolean[n];

        for(int num: nums){
            if(num>0 && num<=n){
                seen[num-1]= true;
            }
        }
        for(int i = 0; i<n;i++){
            if(!seen[i]) return i+1;
        }
        return n+1;

    }
}

File: Data Structures & Algorithms/majority-element-ii/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    majorityElement(nums) {
        const re=[]
        const map = new Map();
        for(let num of nums){
            map.set(num,(map.get(num)||0)+1)
        }
        let k = Math.floor(nums.length/3);
        for(let key of map.keys()){
            if(map.get(key)>k) re.push(key)
        }
        return re
    }
}
